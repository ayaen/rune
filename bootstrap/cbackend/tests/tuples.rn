//  Copyright 2024 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

println ()
println (123,)
println (12, "test")
println (12, ("hello", true))
println (12, ("test", (6.5f64, (~0i1,))))

// indexes
point = (1, 2)
println "point=", point, " point[0]=%u" % point[0], ", point[1]=", point[1]

t = (3*5 + 3 + 5, 3**2 + 5**2)
println t[0], " ", t[1]

// equality
t1 = (1, false)
t2 = (1, false)
println t1, " == ", t2, "? ", t1 == t2
t3 = (1, true)
println t1, " == ", t3, "? ", t1 == t3


/** TODO
// To string
s = ("Alice", 42).toString()
println s
*/