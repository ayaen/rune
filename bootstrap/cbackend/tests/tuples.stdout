()
(123,)
(12, "test")
(12, ("hello", true))
(12, ("test", (6.5, (-1,))))
point=(1, 2) point[0]=1, point[1]=2
23 34
(1, false) == (1, false)? true
(1, false) == (1, true)? false
